docs/site/migration/components/rest-api.md

```diff
@@ -0,0 +1,201 @@
+
+4. Add new REST API endpoints using Express `(req, res, next)` style.
+
+In the following text, we will describe how to migrate these use cases from
+LoopBack 3 style to LoopBack 4.
+
+## General instructions
+
+There are different ways how a LoopBack 3 component can contribute new REST API
+endpoints:
+
+- Some extensions are adding remote methods to existing or newly-created models.
+- Other extensions are contributing Express-style routes.
+
+We recommend extension authors to convert their LoopBack 3 routes to LoopBack 4
+Controller methods wherever possible.
+
+1. Create a new Controller class and add it to your component, see
+   [Contributing REST API endpoints](../../extending/rest-api.md#overview).
+
+2. Move your LoopBack 3 REST API endpoints to the new controller class.
+
+3. Decide if you want these endpoints included in OpenAPI spec. Follow the steps
+   in
+   [Undocumented endpoints](../../extending/rest-api.md#undocumented-endpoints)
+   to hide the new endpoints from the OpenAPI spec describing the target
+   application's API.
+
+4. Optionally, if you want to make your REST endpoints configurable by target
+   applications, then follow the steps in
+   [Dynamic OpenAPI metadata](../../extending/rest-api.md#dynamic-openapi-metadata)
+   to wrap you controller method in a class factory function.
+
+## Migrating REST API endpoints returning HTML pages
+
+To migrate an endpoint that's returning files (or any other non-JSON payload),
+please follow [General instructions](#general-instructions) and then perform two
+additional steps:
+
+1. Modify the controller constructor to inject the HTTP response.
+
+   ```ts
+   import {RestBindings, Response} from '@loopback/rest';
+
+   class MyController {
+     constructor(@inject(RestBindings.Http.RESPONSE) response: Response) {}
+   }
+   ```
+
+2. In the LoopBack 4 controller method, use Express API like `res.contentType()`
+   and `res.send()` to send back the result. (This is a workaround until
+   [loopback-next#436](https://github.com/strongloop/loopback-next/issues/436)
+   is implemented).
+
+## Migrating file upload & download
+
+1. Follow [General instructions](#general-instructions) to create LoopBack 4
+   controllers for your endpoints.
+
+2. Follow the instructions in
+   [Upload and download files ](../../File-upload-download.md) to implement file
+   upload & download functionality in the migrated endpoints.
+
+## Migrating Express routes
+
+We recommend extension authors to use
+[Controllers](https://loopback.io/doc/en/lb4/Controllers.html) as a better
+alternative to low-level Express route API.
+
+If it's not feasible to implement REST endpoints as LoopBack Controllers:
+
+1. Follow the instructions in
+   [Express routes](../../extending/rest-api.md#express-routes) to setup an
+   Express router in your LoopBack 4 component and contribute it to the target
+   application.
+
+2. Take the implementation of Express routes from your LoopBack 3 component and
```

Comment:
Express routes can be added to  a `router` using `router.use(). The `router` can be plugged into LoopBack as an ExpressRequestHandler (express middleware handler function).

Reply:
> Express routes can be added to a router using `router.use()`. 

I am going to clarify this in https://github.com/strongloop/loopback-next/blob/docs/component-migration-rest-api/docs/site/extending/rest-api.md#express-routes

 > The `router` can be plugged into LoopBack as an ExpressRequestHandler (express middleware handler function).

I am not familiar with this new style. For now, I am recommending `app.mountExpressRouter()`, which has the benefit of allowing users to provide OpenAPI spec description of the Express routes too.

I opened https://github.com/strongloop/loopback-next/issues/5423 to update the docs for extension authors.